sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,657 @@ These follow the AAELF64_ base definition from Arm.
+
+The steps above assume that the dynamic linker resolves
+``R_AARCH64_JUMP_SLOT`` relocations in the ``.got.plt`` lazily. The
+``DF_BIND_NOW`` flag in the ``DT_FLAGS`` dynamic tag can be used to
+instruct the dynamic linker to resolve ``R_AARCH64_JUMP_SLOT`` for the
+ELF file containing the tag prior to transferring control to the
+program. When lazy loading is not required the static linker can make
+the ``.got.plt`` relocation-read-only (RELRO).
+
+At run-time the ``LD_BIND_NOW`` environment variable can change the
+dynamic linking behavior. If its value is non-null, the dynamic linker
+evaluates all procedure linkage table entries before transferring
+control to the program. That is, the dynamic linker processes
+relocation entries of type ``R_AARCH64_JUMP_SLOT`` during process
+initialization. Otherwise, the dynamic linker evaluates procedure
+linkage table entries lazily, delaying symbol resolution and
+relocation until the first call to a table entry.
+
+The ``PLT[0]`` entry calls the dynamic linker's entrypoint, stored in
+``.got.plt[2]``, with the following calling convention:
+
++---------------+-----------------------------------------------------------------------------+
+| Location      |    Contents                                                                 |
++===============+=============================================================================+
+| ``ip0``       | address of DL Resolver entry point  ``&.got.plt[2]``                        |
++---------------+-----------------------------------------------------------------------------+
+| ``ip1``       | DL Resolver Entry point                                                     |
++---------------+-----------------------------------------------------------------------------+
+| ``[sp, #0]``  | address of ``.got.plt`` entry for ``PLT[N]`` typically ``&.got.plt[N + 3]`` |
++---------------+-----------------------------------------------------------------------------+
+| ``[sp, #8]``  | ``lr``                                                                      |
++---------------+-----------------------------------------------------------------------------+
+
+Sample PLT sequences
+^^^^^^^^^^^^^^^^^^^^
+
+The following section shows some examples of PLT sequences. The
+sequences assume that the ``.plt.got`` section is within +- 4 GiB from
+the ``.plt`` section.
+
+In the code samples below we assume that the entries in the
+``.got.plt`` are in the same order as the entries in the PLT so that
+code in ``PLT[N]`` loads from ``.got.plt[N + 3]`` where the + 3
+accounts for the 3 reserved entries.
+
+Standard
+~~~~~~~~
+
+This can be used on all Arm platforms when there are no additional
+security features enabled.
+
+PLT header ``PLT[0]``
+
+.. code-block:: asm
+
+    stp    x16, x30, [sp,#-16]!
+    adrp   x16, :page: &.got.plt[2]
+    ldr    x17, [x16, (&.got.plt[2] - &.got.plt[0])]
+    add    x16, x16, (&got.plt[2] - &got.plt[0])
+    br     x17
```

Comment:
I think typo -- I believe `&.got.plt[2] - &.got.plt[0]` should be `:lo12: &.got.plt[2]` (same in the Nth entry below).